azure/services/virtualmachineimages/cache_test.go

```diff
@@ -34,22 +34,22 @@ func TestCacheGet(t *testing.T) {
 		publisher     string
 		offer         string
 		sku           string
-		have          compute.ListVirtualMachineImageResource
+		have          armcompute.VirtualMachineImagesClientListResponse
 		expectedError error
 	}{
 		"should find": {
 			location: "test", publisher: "foo", offer: "bar", sku: "baz",
-			have: compute.ListVirtualMachineImageResource{
-				Value: &[]compute.VirtualMachineImageResource{
+			have: armcompute.VirtualMachineImagesClientListResponse{
+				VirtualMachineImageResourceArray: []*armcompute.VirtualMachineImageResource{
```

Comment:
Just curious, is there a reason why this is a list of pointers instead of a normal list? I'm not sure how there would be a situation where you have nil list entries.

Reply:
cc @mboersma Since he implemented this part. I'm not actually entirely sure

Reply:
Fair enough. I don't expect you to change anything especially if it's a type from the SDK, but I just wonder if we're overusing pointers sometimes.

Reply:
It's a type change they made basically *everywhere* in SDKv2: if a field used to be a pointer to a slice, now it's a slice of pointers. 🤷🏻